dbusmock/templates/accounts_service.py

```diff
@@ -0,0 +1,392 @@
+        ('SetRealName', 's', '', SetRealName),
+        ('SetEmail', 's', '', SetEmail),
+        ('SetLanguage', 's', '', SetLanguage),
+        ('SetXSession', 's', '', SetXSession),
+        ('SetSession', 's', '', SetSession),
+        ('SetSessionType', 's', '', SetSessionType),
+        ('SetLocation', 's', '', SetLocation),
+        ('SetHomeDirectory', 's', '', SetHomeDirectory),
+        ('SetShell', 's', '', SetShell),
+        ('SetIconFile', 's', '', SetIconFile),
+        ('SetLocked', 'b', '', SetLocked),
+        ('SetAccountType', 'i', '', SetAccountType),
+        ('SetPasswordMode', 'i', '', SetPasswordMode),
+        ('SetPasswordHint', 's', '', SetPasswordHint),
+        ('SetPassword', 'ss', '', SetPassword),
+        ('SetAutomaticLogin', 'b', '', SetAutomaticLogin),
+        ('GetPasswordExpirationPolicy', '',
+         'xxxxxx', GetPasswordExpirationPolicy),
+    ])
+
+    had_users = len(self.mock_users) != 0
+    had_multiple_users = len(self.mock_users) > 1
+    user = mockobject.objects[path]
+    user.password = 'Pa$$wo0rd'
+    user.properties = default_props
+    user.pwd_expiration_policy = {
+        'expiration_time': 1000,
+        'last_change_time': 1615832413,
+        'min_days_between_changes': 0,
+        'max_days_between_changes': 0,
+        'days_to_warn': 0,
+        'days_after_expiration_until_lock': 0,
+    }
+    self.mock_users[uid] = default_props
+
+    self.EmitSignal(MAIN_IFACE, 'UserAdded', 'o', [path])
+
+    if not had_users:
+        emit_properties_changed(self, MAIN_IFACE, 'HasNoUsers')
+    elif not had_multiple_users and len(self.mock_users) > 1:
+        emit_properties_changed(self, MAIN_IFACE, 'HasMultipleUsers')
+
+    return path
+
+
+@dbus.service.method(MOCK_IFACE, in_signature='', out_signature='ao')
+def ListMockUsers(self):
+    """ List the mock users that have been created """
+    return [get_user_path(uid) for uid in self.mock_users.keys()]
+
+
+@dbus.service.method(MOCK_IFACE, in_signature='s', out_signature='o')
+def AddAutoLoginUser(self, username):
+    """ Enable autologin for an user """
+    path = self.FindUserByName(username)
+    self.automatic_login_users.append(path)
+    mockobject.objects[path].SetAutomaticLogin(True)
+    emit_properties_changed(self, MAIN_IFACE, 'AutomaticLoginUsers')
+    return path
+
+
+@dbus.service.method(MOCK_IFACE, in_signature='s', out_signature='o')
+def RemoveAutoLoginUser(self, username):
+    """ Disables autologin for an user """
+    path = self.FindUserByName(username)
+    self.automatic_login_users.remove(path)
+    mockobject.objects[path].SetAutomaticLogin(False)
+    emit_properties_changed(self, MAIN_IFACE, 'AutomaticLoginUsers')
+    return path
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='ssi', out_signature='o')
+def CreateUser(self, name, fullname, account_type):
+    """ Creates an user using the default API """
+    try:
+        self.FindUserByName(name)
+        found = True
```

Comment:
Please raise the exception directly here and eliminate `found`.

Reply:
Eh, I try to do that previously, but the problem here is that I want to proceed if the exception is raised, so if I raise a new one here I'll be caught by the `except dbus.exceptions.DBusException` later on, so... I'd never be able to handle this.

I was looking to some clearner way to do this but didn't get anything better than this

Reply:
Heh, right, nevermind then.